zipline/algorithm.py

```diff
@@ -190,6 +190,18 @@ def __init__(self, *args, **kwargs):
 
         self.instant_fill = kwargs.pop('instant_fill', False)
 
+        # If an env has been provided, pop it
+        self.trading_environment = kwargs.pop('env', None)
+
+        if self.trading_environment is None:
```

Comment:
TradingEnvironments can either be provided as a kwarg, or a default is built by the TradingAlgorithm.